aips/aip-113.md

```diff
@@ -45,67 +50,66 @@ Derivable AA introduces the following features:
+        // then authentication_key to hex, and then we need separators as well - like ::
+        let bytes = bcs::to_bytes(&derivable_func_info);
+        bytes.append(bcs::to_bytes(abstract_public_key));
+        bytes.push_back(DERIVABLE_ABSTRACTION_DERIVED_SCHEME);
+        from_bcs::to_address(hash::sha3_256(bytes))
+    }
 ```
 
-
 ### Authentication Flow
 Derivable AA shares the same flow with normal AA in native code but diverge when it comes to move code. When calling `0x1::account_abstraction::authenticate`, if the `auth_data` is using Derivable AA, 
-1. The `account_identity` in the `auth_data` is used to derive the account address.
+1. The `abstract_public_key` in the `auth_data` is used to derive the account address.
 2. The derived address is checked against the domain’s whitelist to verify if the function_info exists in the domain’s registered functions.
 3. If the identity matches and the function exists, the account is authenticated by the corresponding function as normal AA.
 ```rust
-fun authenticate(
-    account: signer,
-    func_info: FunctionInfo,
-    signing_data: AbstractionAuthData,
-): signer acquires DispatchableAuthenticator, DomainDispatchableAuthenticator {
-    let master_signer_addr = signer::address_of(&account);
-
-    if (signing_data.is_domain()) {
-        let func_infos = &borrow_global<DomainDispatchableAuthenticator>(@aptos_framework).auth_functions;
-        assert!(func_infos.contains(&func_info), error::not_found(EFUNCTION_INFO_EXISTENCE));
-        assert!(master_signer_addr == domain_aa_account_address(func_info, signing_data.account_identity()), error::invalid_state(EINCONSISTENT_SIGNER_ADDRESS));
-    } else {
-        assert!(using_dispatchable_authenticator(@aptos_framework), error::not_found(EDISPATCHABLE_AUTHENTICATOR_IS_NOT_USED));
-        let func_infos = &borrow_global<DispatchableAuthenticator>(resource_addr(master_signer_addr)).auth_functions;
-        assert!(ordered_map::contains(func_infos, &func_info), error::not_found(EFUNCTION_INFO_EXISTENCE));
-    };
-
-    function_info::load_module_from_function(&func_info);
-    let returned_signer = dispatchable_authenticate(account, signing_data, &func_info);
-    assert!(master_signer_addr == signer::address_of(&returned_signer), error::invalid_state(EINCONSISTENT_SIGNER_ADDRESS));
-    returned_signer
-}
+    fun authenticate(
+        account: signer,
+        func_info: FunctionInfo,
+        signing_data: AbstractionAuthData,
+    ): signer acquires DispatchableAuthenticator, DerivableDispatchableAuthenticator {
+        let master_signer_addr = signer::address_of(&account);
+
+        if (signing_data.is_derivable()) {
+            assert!(features::is_derivable_account_abstraction_enabled(), error::invalid_state(EDERIVABLE_ACCOUNT_ABSTRACTION_NOT_ENABLED));
+            assert!(master_signer_addr == derive_account_address(func_info, signing_data.derivable_abstract_public_key()), error::invalid_state(EINCONSISTENT_SIGNER_ADDRESS));
```

Comment:
and here we should check for key rotation?
(we can implement later, but just checking if it makes sense as a concept)
```move
let auth_key = derive_auth_key(func_info, signing_data.derivable_abstract_public_key());
let account_address = lookup_original_account_address(auth_key);
assert!(master_signer_addr == account_address, error::invalid_state(EINCONSISTENT_SIGNER_ADDRESS));
```

Reply:
hmmm, not sure we want rotation as a concept - wallet would not be able to know/derive the old address. but if we do - we can have that as a separate AIP